scripts/build-post-list.ts

```diff
@@ -47,10 +51,12 @@ export function slugifyToC(str: string) {
 }
 
 /**
- * Capitalizes the first letter of each word in a string.
+ * Capitalizes the first letter of each word in the provided string.
+ *
+ * This function splits the text on whitespace and hyphen characters, capitalizes the first letter of each segment, and then joins them with a space.
  *
- * @param {string} text - The string to capitalize.
- * @returns {string} - The capitalized string.
+ * @param text - The string to transform.
+ * @returns The transformed string with each word's initial letter capitalized.
```

Comment:
Should add the return type here as it is not explicitly mentioned in the function definition.

More better way will be to define the return type in the function definition itself, and the JSDoc comment (@return) will itself be able to describe what is being returned.